src/components/Carousel.tsx

```diff
@@ -0,0 +1,111 @@
+'use client';
+
+import { merge } from '@/lib/utils';
+import { type EmblaOptionsType } from 'embla-carousel';
+import Autoplay from 'embla-carousel-autoplay';
+import useEmblaCarousel from 'embla-carousel-react';
+import { Play } from 'lucide-react';
+import React, { useEffect } from 'react';
+
+type CarouselProps = {
+	delay?: number;
+	children: SlideElement[];
+};
+
+export const Carousel = ({
+	children,
+	delay = 3000,
+	...rest
+}: CarouselProps & EmblaOptionsType) => {
+	const slidesLength = children
+		.map((child) => (child.props.full ? 2 : 1))
+		.reduce((a, b) => a + b, 0);
+
+	const [emblaRef, emblaApi] = useEmblaCarousel(
+		{
+			align: 'start',
+			// Makes sure full slides are not shown only partially
+			slidesToScroll: 'auto',
+			// Makes sure off screen slides are not counted if using emblaApi.slidesInView()
+			inViewThreshold: 0.1,
+			...rest,
+		},
+		// Autoplay only if slides are over a page long in total
+		slidesLength > 2 ? [Autoplay({ delay })] : undefined
+	);
+
+	// Show play button when autoplay is stopped
+	const [autoplay, setAutoplay] = React.useState(true);
+	useEffect(() => {
+		if (!emblaApi) return;
+		emblaApi.on('autoplay:stop', () => setAutoplay(false));
+		emblaApi.on('autoplay:play', () => setAutoplay(true));
+	}, [emblaApi]);
+
+	return (
+		<>
+			{/* Carousel */}
+			<div
+				className="min-w-full flex-auto overflow-hidden"
+				ref={emblaRef}
+			>
+				<div className={`mx-6 -mr-0 flex h-full py-6`}>{children}</div>
+			</div>
+			{/* Continue autoplay button */}
+			<div
+				className={`${autoplay ? 'invisible' : 'visible'} absolute bottom-8 right-8 z-50 rounded bg-white/0 p-2 hover:bg-white/30`}
+				onClick={() => {
+					if (emblaApi) emblaApi.plugins().autoplay?.play();
```

Comment:
prefer to extract this to a function definition before `return` (`const resumeAutoplay = ...`, `onClick={resumeAutoplay}`)